frontend/src/components/Classroom/ClassroomList.vue

```diff
@@ -0,0 +1,135 @@
+<script setup>
+import { onMounted, ref, watch } from 'vue';
+import classroomService from '@/api/classrooms';
+import DeleteIcon from '@/assets/img/delete_icon.svg';
+import { parseDates } from '@/helpers/parse';
+
+const classrooms = ref(null);
+const searchTerm = ref('');
+const fields = {
+  name: 'Name',
+  capacity: 'Capacity',
+  createdAt: 'Created At',
+};
+
+onMounted(() => filterClassrooms(''));
+watch(searchTerm, newSearchTerm => filterClassrooms(newSearchTerm));
+const filterClassrooms = (search = '') => {
+  classroomService.getAllClassrooms().then(resp => {
+    const parsed = parseDates(resp);
+    const filtered = parsed.filter(room => room.name.toLowerCase().includes(search.toLowerCase()));
+    classrooms.value = filtered;
+  });
+};
+const even = num => (num % 2 === 0 ? '' : 'rsprd-table__row-darker');
+const handleDelete = async classroomId => {
+  const isDeleteConfirmed = confirm('Do you really want to delete classroom?');
+  if (!isDeleteConfirmed) return;
+
+  const deleteResponse = await classroomService.deleteClassroomById(classroomId);
+
+  if ('error' in deleteResponse) {
+    return alert('Internal Server Error. Can not get classroooms.');
+  }
+
+  filterClassrooms(searchTerm.value);
+};
+</script>
+
+<template>
+  <div class="main">
+    <div class="rsprd-bar">
+      <h2 class="rsprd-bar__title">Classrooms</h2>
+      <input v-model="searchTerm" class="rsprd-body__input" type="text" placeholder="Search..." />
+    </div>
+    <hr />
+    <div class="rsprd-body">
+      <table class="rsprd-table">
+        <thead>
+          <tr class="rsprd-table__heading rsprd-table__row">
+            <th v-for="(value, key) in fields" :key="key" class="rsprd-table__cell">{{ value }}</th>
+            <th class="rsprd-table__cell">Action</th>
+          </tr>
+        </thead>
+        <tbody>
+          <tr
+            v-for="(classroom, index) in classrooms"
+            :key="classroom.id"
+            class="rsprd-table__row"
+            :class="even(index)"
```

Comment:
I'd rewrite to something like this:
```
:class="{ 'rsprd-table__row-darker': index % 2 === 0 }"
```
since this logic is only for visual part of the component I think best place for it to be is in template